src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRequest.cs

```diff
@@ -37,6 +37,9 @@ public class RestoreRequest
 
             // Default to the project folder
             RestoreOutputPath = Path.GetDirectoryName(Project.FilePath);
+
+            // Limit concurrency based on the number of processors.
+            MaxDegreeOfConcurrency = Math.Min(Environment.ProcessorCount, DefaultDegreeOfConcurrency);
```

Comment:
This is still not the complete fix, Since RestoreArgs at https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs#L207 update it back to either 1 or `DefaultDegreeOfConcurrency` which decides how many concurrent restore jobs run.

Reply:
:+1: fixed